geochem_inverse_optimize.py

```diff
@@ -281,6 +280,279 @@ def get_roughness(self) -> float:
         return cp.norm(cp.vstack(self._regularizer_terms)).value
 
 
+class SampleNetworkContinuous:
+
+    # TODO: Docstrings
+
+    def __init__(self, sample_network: nx.DiGraph, area_labels: np.array, nx: int, ny: int) -> None:
+        self.sample_network = sample_network
+        self.grid = InverseGrid(nx, ny, area_labels, sample_network)
+        self._site_to_parameter: Dict[str, ReciprocalParameter] = {}
+        self._primary_terms = []
+        self._regularizer_terms = self._build_regularizer_terms()
+        self._regularizer_strength = cp.Parameter(nonneg=True)
```

Comment:
@r-barnes Are you suggesting that this just extends `SampleNetwork` but we now include a "continuous" bool flag in the `__init__` similar to the optional regularization. I think it should be fairly doable and I'll work to that for now.

Reply:
@AlexLipp : No, sorry.

Reply:
I'm saying that `SampleNetwork` has a boolean to determine if regularization is enabled and this class should also have that flag. Ideally we want to avoid any non-essential differences, especially in the interface, and then think hard about how how to simplify the real differences.

Reply:
I think my preferred method, as mentioned ... elsewhere? ... is to set up a sample network (using networkx, &c) to represent the continuous case and then have the existing `SampleNetwork` class ingest that. But if that looks terrible, then we can do something similar to what you have here.

Reply:
Got it. Sorry I actually figured this out separately and deleted the above comment before seeing your reply so this thread seems a little disjointed, but its resolved now as you suggested. 

> I think my preferred method, as mentioned ... elsewhere? ... is to set up a sample network (using networkx, &c) to represent the continuous case and then have the existing SampleNetwork class ingest that. But if that looks terrible, then we can do something similar to what you have here.

I'm not entirely sure what you mean by this, but how I'm interpreting is just trying to have everything as a single `SampleNetwork` class, perhaps with a `continuous` flag to indicate to solve discretely or not. I will try and set this up.